src/containers/AdminIncomingMessageList/components/IncomingMessageFilter.tsx

```diff
@@ -202,6 +211,19 @@ const IncomingMessageFilter: React.FC<IncomingMessageFilterProps> = (props) => {
     getTexters?.organization?.memberships?.edges ?? []
   );
 
+  useEffect(() => {
+    if (!isEmpty(campaignOptions) && props.campaignId) {
+      setCampaignOption(
+        campaignOptions.find((campaign) => campaign.id === props.campaignId)
+      );
+    }
+    if (!isEmpty(texterOptions) && props.texterId) {
+      setTexterOption(
+        texterOptions.find((texter) => texter.id === props.texterId)
+      );
+    }
+  }, [getCampaigns, getTexters]);
```

Comment:
If we used just Campaign IDs, we’d have to use an Array for the options and a map for the lookup. Should be easy to implement. I’ll see if there are any other issues while trying to implement it, otherwise will go ahead and implement that.